app/src/main/java/com/example/courses/presentation/viewModel/MainViewModel.kt

```diff
@@ -36,23 +53,53 @@ class MainViewModel() : ViewModel() {
     }
 
     fun take(): Person {
-        return sharedPreferencesUseCase.take()
+        val person = sharedPreferencesUseCase.take()
```

Comment:
I'd prefer to add different naming for sharedPreferencesUseCase. Use Case answers for a question "What to take?" I'd call it "PersonsCreationProgressUseCase", or "PersonInputUseCase" or more common "InputUseCase". Your naming answers to a question "Where to take?".